### DISCUSSION TITLE
General Development Suggestions (mostly UI)

### DISCUSSION THREAD

**Discussion by panthen** [GrandOrgue/grandorgue#812] [General]
Apologies for the long wall of text. Most of these suggestions are generally aimed at improving user experience and professionalising the look of GO. 

The perennial issues – scalable interface now solved (thank you, Oleg!) - of development of a swell box model beyond amplitude, and an improved tremulant model are the obvious major features that development should focus on, but I feel that the following would greatly improve the look and feel of GO (particularly the start menu).

Also, I appreciate that there's quite a lot here, and I have _no idea if _any_ of it is remotely feasible_. If designing a new look requires no coding experience, I am happy to give it a go if someone can point me in the right direction.
Regardless, none of this is a criticism of GO - the program is a marvel, and I am in awe (and greatly appreciative) of the time and effort that has been devoted to it.

1.	Start Menu – akin to Microsoft word, Sibelius, Musescore, etc.
o	See picture for a mock-up layout (this is not meant to look good or ‘realistic’, it is just a guide). If you would like me to put an actual possible design together rather than a quick thing in Inkscape, I can give it a go.
o	Makes initial navigation much simpler – no fumbling through drop-down menus
-Recently used organs are one click away from loading
     -Includes access to an organ library of installed organs: see point 2
     -Organ Setup button easily accessible 
     -Audio options
     -Install new organ (See 8)

2.	Organ Library :
o	In-built alternative to file explorer.
o	Searchable
o	Organs are shown with thumbnail pictures
     -I suggest that all Piotr Grabowski’s organs all have pre-chosen pictures
     -By necessity (I assume) this will have to be user chosen. To change, right-click on a temporary image [the logo?].
o	Organs can be given customisable sorting tags e.g.: No. of manuals, Favourited, Practice, Baroque, Neo-Baroque, Chamber, French Romantic…
o This would replace the 'load' option in the menu - essentially prettifying it. 
o 1 & 2 will involve corresponding options under the 'file' drop down. Thus removing the confusing and overlapping: 'load', 'favourites', 'open', 'open recent', and 'install organ package': we only need recent, library, & install.

3.	Auto-update function or check for updates button

4.	Metronome:
o	Volume adjustment (if it’s there, it’s too hard to find)
o	Alternative sounds: particularly wood block

5.	Higher resolution default background textures: for demo-organ and for odf’s built with odf builder.
o	New square button option instead of circular stops – makes it easier to layout and read text on the stop
o	Automatic stop panel generation – I’m not yet sure how to create a panel with just e.g. the left or right jamb without the centre organ console. It would be good if GO could do this automatically where a sampleset creator has not created their own.

6.	I know that Lars has previously opposed this, but I would suggest adding progressing/regressing buttons to the piston panel. You can do this with the combination setter, but this is more cumbersome, and I would rather divisional/general pistons to be midi controlled. Introducing this would not require any changes to the combination setter, but means that everybody can use GO how they want to.

7.	Remove organ packages – I believe this is a legacy thing? It’s a bit confusing with users self-installing organs into the documents folder – when I first used GO I put my organ ‘packages’ (uncompressed zipped files) into here as this was where it sounded like it wanted it prior to installing the organ.

8.	Continuing on I suggest that GO installs the organs into the directory itself rather than the user
o	i.e. 1. I download a folder from say Piotr Grabowski; 2. from within GO I select install organ. 3. GO uncompresses the folder 
     and transfers the files into its directory. 4. GO asks the user whether they would like to delete the downloaded files (this could be toggleable in settings). 5. GO asks the user to select a thumbnail (auto direct to image folder that are in Piotr’s Data folder)  6. Organ install success. 7. Possibly ask to generate cache?

9.	Automatically save Piston/Combination settings on exit.
o	It would be nice if this could be saved separately from the stops as I currently have to deselect all stops/couplers to do this. If it can be done already, could I suggest it being made more obvious as I don't have a clue.

10.	A low priority addition would be compressor and eq functions - generally superfluous, but nice to have.

11.	Also in my opinion (as a low priority) the top menu bar should eventually be removed and replaced with a menu that appears when clicking on a button. I can’t think of any examples from other programs, but the less clutter on the screen the better – more space can be given to showing the stop jambs etc. Essentially I don't alter the reverb length, polyphony count/meter, transpose, or volume with any regularity, and don't feel it needs to be on the screen all the time.

Edit: 2 more suggestions:
12. That the bit rate of samples can be set per organ (allowing larger organs to be loaded in, say, 16 bit definition, and smaller organs in 24 bit. This is too convoluted at the moment, and if you forget to set the settings back before you load another organ it's going to create a new cache.
13. A small visual change would be to have a new organ loading graphic rather than the (in my case) windows default green scrolling bar. An example could be a greyed out organ being filled in, or the new GO icon being filled in, or a set of pipes where each pipe swells in series. Accompanying text could describe what division is being loaded.
14. It could also encourage sampleset producers to supply odfs if the organ builder project was renewed. It's not too far off, but there are some important functions missing which means you have to learn the system and use notepad++. Creating odfs mostly isn't too challenging - though I haven't yet got the hang of creating panels - but finding out how to do it was difficult, as I had to piece together various things from various different places.

Second Edit:
15. After loading an organ for the first time, a walkthrough Midi set-up should be displayed asking the user to press the device they intend on using. This could be disabled in settings, and normal right-click to set would still be there. 
o Alternatively, would it be possible to do this on installing GO, that the user is asked how many manuals they wish to set up and then are asked to play them in series guided by visuals.
o Also there should be a way of improving 'stop' set-up. I use a launchpad, and it is a _pain_ having to right click on each stop and then changing it to 9x note on toggle. Could this not be replaced by a set-up on a sampleset's first load which would allow for setting events for the device used, and then highlighting each stop as the user responds on his/her midi device?
o Clicking on each one is an unnecessary pain. Even a simple 'next' button would be a relief!

 
![menu](https://user-images.githubusercontent.com/92125704/139504159-4c14097b-4d3b-4bb8-bc69-7e0ca6a3a949.png)

**Comment by vpoguru** [comment]
Very good points!

**Reply by panthen** [reply]
Thank you. Hopefully they can stimulate a useful discussion - I don't pretend to have all the answers and I don't have a clue what is actually possible to develop!

**Comment by panthen** [comment]
As an addition, Mark's (Bumblebee001) recent post made me look back over Piotr Grabowski's reasons for no longer producing odfs for his samplesets. Whilst I'm sure the devs know what improvements to make, answering his criticisms seems a good place to start.

His summary is that GO is not powerful enough and too hard to use.
Piotr Grabowski stopped making odfs as:
- organ installation is too difficult (see the above for my suggestion)
- patching is difficult (I can't comment on this, wouldn't it just require copying over new audio files/odf?)
- UI scaling (Oleg, I think, fixed this?) 
- swell pedal noise (would this just require playing back samples based off the speed at which the MIDI data is sent through?)
- non-standard combination action (I'm not sure, could this be implemented by an additional switch system in the odf?)
- using temperaments resets samples to 440Hz (are the temperaments only coded to work from a set point?)
- Crossfade between the sustaining part and silence is too short (re: percussive samples - is it possible to implement specific crossfade lengths for different sample types?)

**Reply by rousseldenis** [reply]
Yes, this was his last answer :
https://www.facebook.com/PiotrGrabowskiSampleSets/posts/322095949719993

So, he said 'maybe' he will go back to GO.

Indeed, as his productions are the best ones (for my point of view), we should try to hear his requirements in order to get him back.

**Reply by Jean-Gabriel1982** [reply]
Hello all,

Regarding all the suggestions in this thread, I think GO must remain "clean, light, efficient". I know too much people complaining in HW too heavy program. 

Regarding UI, we must not forget that in the end, the user only sees the stops panels. 

I think that it would then be necessary, for the configuration of the organ, a window adapted to the definition of the screen (and not a floating palette as currently), which offers at a glance all the tools necessary to user to configure their organ.

I would even dare to say that to install or configure the organ and all that goes with it, the user must not say to himself "how do we do that?" but let him say "I am doing this".

on the model of icon proposals, perhaps we could ask the community for interface proposals, which go in this direction?

finally, as it had already been discussed, to bring in, or to return organ creators, perhaps it would be good to reflect, with the community, on the structure of the ODF files?

**Reply by jason-barba** [reply]
I second this.

**Reply by BasKlb** [reply]
Using icons to load Organs into GO is already possible outside of GO itself. I've been doing this for quite some time. GrandOrgue's menu bar is very small on my 10.1 InchTouchscreen (resolution 1280x800) and is almost impossible to operate with touch. This is especially true for the list of organs.
That's why I load my samplesets via shortcuts to the .organ Odf files and I gave those shortcuts their own icon for each organ. By creating multiple desktops it was also easy to group the shortcuts. This way I don't use the cache to speed up the loading, but that is not a must with the current fast NVMe M2 SSDs.
It works fine like this. You just have to be careful to close an organ if you want to switch to another organ
![Organs screen](https://user-images.githubusercontent.com/95912473/145578507-57fbd51d-146d-45af-8614-b7c4f423e493.png)
